app/services/import/legacy/transform-companies.service.js

```diff
@@ -0,0 +1,64 @@
+'use strict'
+
+/**
+ * Transforms NALD company data into a valid object that matches the WRLS structure
+ * @module ImportLegacyTransformCompaniesService
+ */
+
+const FetchCompanyService = require('./fetch-company.service.js')
+const CompanyPresenter = require('../../../presenters/import/legacy/company.presenter.js')
+const ImportCompanyValidator = require('../../../validators/import/company.validator.js')
+
+/**
+ * Transforms NALD company data into a validated object that matches the WRLS structure
+ *
+ * @param {string} regionCode - The NALD region code
+ * @param {string} licenceId - The NALD licence ID
+ *
+ * @returns {Promise<object>} an object representing a valid WRLS company
+ */
+async function go (regionCode, licenceId) {
+  const companies = await FetchCompanyService.go(regionCode, licenceId)
+
+  const transformedCompanies = []
+
+  companies.forEach((company) => {
+    const transformedCompany = CompanyPresenter.go(company)
+
+    ImportCompanyValidator.go(transformedCompany)
+
+    transformedCompanies.push(transformedCompany)
+  })
+
+  return {
+    transformedCompanies: _removeDuplicateCompanies(transformedCompanies),
+    companies
+  }
+}
+
+/**
+ * Remove duplicate companies from the transformed array
+ *
+ * We need to get the address and party id for each company.
+ *
+ * But some of these companies will share the same party id. We can remove these duplicates
+ * and only persist the company once.
+ *
+ * @param {object[]} transformedCompanies - The transformed companies
+ *
+ * @param transformedCompanies
+ * @returns {object[]} -The transformed companies array with no duplicates
+ */
+function _removeDuplicateCompanies (transformedCompanies) {
```

Comment:
I think we remove duplicates for now. 

We might be able to save the addresses etc based on the simplified list of companies. 

We have the party code, might need to get the region when it comes to it but i think it should work this way.

Reply:
We will get all the party id's (company id's) and transform them into a set. 

This array can then be used to fetch the addresses / contacts etc in other transform services.